cmd/fetcher/main.go

```diff
@@ -234,10 +240,98 @@ func run(ctx context.Context, root string) ([]createdPlugin, error) {
 	return created, nil
 }
 
+func getLatestBaseImageVersions(basedir string) (_ map[string]string, retErr error) {
+	// Walk up from plugins dir to find .github dir
+	rootDir := basedir
+	var githubDir string
+	for {
+		githubDir = filepath.Join(rootDir, ".github")
+		if st, err := os.Stat(githubDir); err == nil && st.IsDir() {
+			break
+		}
+		newRootDir := filepath.Dir(filepath.Dir(githubDir))
+		if newRootDir == rootDir {
+			return nil, fmt.Errorf("failed to find .github directory from %s", basedir)
+		}
+		rootDir = newRootDir
+	}
+	dockerDir := filepath.Join(githubDir, "docker")
+	d, err := os.Open(dockerDir)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		retErr = errors.Join(retErr, d.Close())
+	}()
+	entries, err := d.ReadDir(-1)
+	if err != nil {
+		return nil, err
+	}
+	latestVersions := make(map[string]string, len(entries))
+	for _, entry := range entries {
+		if !strings.HasPrefix(entry.Name(), "Dockerfile.") {
+			continue
+		}
+		imageName, version, err := parseDockerfileBaseImageNameVersion(filepath.Join(dockerDir, entry.Name()))
+		if err != nil {
+			return nil, err
+		}
+		latestVersions[imageName] = version
+	}
+	return latestVersions, nil
+}
+
+func parseDockerfileBaseImageNameVersion(dockerfile string) (_ string, _ string, retErr error) {
+	f, err := os.Open(dockerfile)
+	if err != nil {
+		return "", "", nil
+	}
+	defer func() {
+		retErr = errors.Join(retErr, f.Close())
+	}()
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		line := strings.TrimSpace(s.Text())
+		fields := strings.Fields(line)
+		if len(fields) < 2 {
+			continue
+		}
+		if !strings.EqualFold(fields[0], "from") {
+			continue
+		}
+		var image string
+		for i := 1; i < len(fields); i++ {
+			if strings.HasPrefix(fields[i], "--") {
+				// Ignore --platform and other args
+			}
+			image = fields[i]
+			break
+		}
+		if image == "" {
+			return "", "", fmt.Errorf("missing image in FROM: %q", line)
+		}
+		imageName, version, found := strings.Cut(image, ":")
+		if !found {
+			return "", "", fmt.Errorf("invalid FROM line: %q", line)
+		}
+		return imageName, version, nil
+	}
+	if err := s.Err(); err != nil {
+		return "", "", err
+	}
+	return "", "", fmt.Errorf("failed to detect base image in %s", dockerfile)
+}
+
 // copyDirectory copies all files from the source directory to the target,
 // creating the target directory if it does not exist.
 // If the source directory contains subdirectories this function returns an error.
-func copyDirectory(source, target, prevVersion, newVersion string) (retErr error) {
+func copyDirectory(
+	source string,
+	target string,
+	prevVersion string,
+	newVersion string,
+	latestBaseImageVersions map[string]string,
```

Comment:
At some point I'd like to refactor this to run on a struct so we don't have to plumb arguments all the way down through.